src/pages/SignupPage/FormElement.tsx

```diff
@@ -0,0 +1,27 @@
+import { FieldValues, Path, UnPackAsyncDefaultValues, UseFormRegister } from 'react-hook-form';
+
+interface IFormElementProps<T extends FieldValues> {
+  register: UseFormRegister<T>;
+  message?: string;
+  name: Path<UnPackAsyncDefaultValues<T>>;
+  main: string;
+}
+export const FormElement = <T extends FieldValues>(props: IFormElementProps<T>) => {
+  const { register, message, name, main } = props;
+  return (
+    <>
+      <div className="label-input">
+        <label htmlFor={main}>{main}</label>
+        <br />
+        <input
+          type="text"
+          id={main}
+          placeholder={main}
+          className={message ? ' data-from-user with-errors' : 'data-from-user'}
+          {...register(name)}
+        />
```

Comment:
<label>
  <span>
  <input>
</label>

and you don't need htmlFor and id

and please remove br